### PR TITLE
[stable/redmine] Simplify ingress configuration for cert-manager

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juan@bitnami.com>

#### What this PR does / why we need it:

This PR allows the user to configure the proper annotations when using cert-manager to handle TLS certificates in an easy way. The user just needs to setup a boolean to true to do so

### CODE REVIEW

File: stable/redmine/templates/ingress.yaml [hunk 1]
```diff
@@ -1,23 +1,37 @@
 {{- if .Values.ingress.enabled }}
-apiVersion: {{ required "A valid .Values.networkPolicyApiVersion entry required!" .Values.networkPolicyApiVersion }}
+{{- range .Values.ingress.hosts }}
+apiVersion: extensions/v1beta1
```
Comment (tompizmor) [inline stable/redmine/templates/ingress.yaml:3]: you are not using the `.Values.networkPolicyApiVersion` value any more...is that right?

File: stable/redmine/templates/ingress.yaml [hunk 2]
```diff
@@ -1,23 +1,37 @@
...
 kind: Ingress
 metadata:
-{{- if .Values.ingress.annotations }}
+  name: "{{- printf "%s-%s" .name $.Release.Name | trunc 63 | trimSuffix "-" -}}"
+  labels:
+    app: {{ template "redmine.fullname" $ }}
+    chart: "{{ $.Chart.Name }}-{{ $.Chart.Version }}"
```
Comment (tompizmor) [inline stable/redmine/templates/ingress.yaml]: shouldn't you use `{{ template "redmine.chart" $ }}` ?

File: stable/redmine/values.yaml [hunk 3]
```diff
@@ -143,16 +143,36 @@ serviceType: LoadBalancer
...
+  ## The list of hostnames to be covered with this ingress record.
+  ## Most likely this will be just one host, but in the event more hosts are needed, this is an array
+  hosts:
+  - name: redmine.local
+
+    ## Set this to true in order to enable TLS on the ingress record
+    ## A side effect of this will be that the backend wordpress service will be connected at port 443
+    tls: false
+
+    ## Set this to true in order to add the corresponding annontations for cert-manager
```
Comment (tompizmor) [inline stable/redmine/values.yaml]: typo

**Review by tompizmor — CHANGES_REQUESTED** [review]

File: stable/redmine/templates/ingress.yaml (hunk 2 above)
Comment (juan131) [reply stable/redmine/templates/ingress.yaml]: Done

File: stable/redmine/templates/ingress.yaml (hunk 1 above)
Comment (juan131) [reply stable/redmine/templates/ingress.yaml:3]: I don't know why we were setting this..

Comment (tompizmor) [reply stable/redmine/templates/ingress.yaml:3]: If you don't know maybe you should leave it as it was before

Comment (davidkarlsen) [reply stable/redmine/templates/ingress.yaml:3]: .Capabilitties.ApiVersion can be used for probing what versions are supported: https://github.com/helm/helm/blob/master/docs/chart_template_guide/builtin_objects.md - the guidelines say to support the two latest minor versions of k8s.

Comment (juan131) [reply stable/redmine/templates/ingress.yaml:3]: Great suggestion @davidkarlsen 

However, in this case I don't know if it's worth. We're just setting the **apiVersion** for an Ingress resource which is supported on **extensions/v1beta1** for every K8s version

Comment (tompizmor) [reply stable/redmine/templates/ingress.yaml:3]: I am fine with this but delete the `networkPolicyApiVersion` references also from values.yaml and README

Comment (juan131) [reply stable/redmine/templates/ingress.yaml:3]: Done @tompizmor

**Pull request merged**